packages/libs/lambda-at-edge/src/lib/triggerStaticRegeneration.ts

```diff
@@ -0,0 +1,70 @@
+import { s3BucketNameFromEventRequest } from "../s3/s3BucketNameFromEventRequest";
+import { buildS3RetryStrategy } from "../s3/s3RetryStrategy";
+import { OriginRequestDefaultHandlerManifest } from "../types";
+
+interface TriggerStaticRegenerationOptions {
+  request: AWSLambda.CloudFrontRequest;
+  response: AWSLambda.CloudFrontResponse;
+  manifest: OriginRequestDefaultHandlerManifest;
+  basePath: string | undefined;
+}
+
+export const triggerStaticRegeneration = async (
+  options: TriggerStaticRegenerationOptions
+): Promise<void> => {
+  const { region } = options.request.origin?.s3 || {};
+  const bucketName = s3BucketNameFromEventRequest(options.request);
+
+  const { SQSClient, SendMessageCommand } = await import("@aws-sdk/client-sqs");
+  const sqs = new SQSClient({
+    region,
+    maxAttempts: 3,
+    retryStrategy: await buildS3RetryStrategy()
+  });
+
+  const lastModifiedAt = new Date(
+    options.response.headers["last-modified"]?.[0].value
+  )
+    .getTime()
+    .toString();
+
+  await sqs.send(
+    new SendMessageCommand({
+      QueueUrl: `https://sqs.${region}.amazonaws.com/${bucketName}.fifo`,
+      MessageBody: options.request.uri, // This is not used, however it is a required property
+      MessageAttributes: {
+        BucketRegion: {
+          DataType: "String",
+          StringValue: region
+        },
+        BucketName: {
+          DataType: "String",
+          StringValue: bucketName
+        },
+        CloudFrontEventRequest: {
+          DataType: "String",
+          StringValue: JSON.stringify(options.request)
+        },
+        Manifest: {
+          DataType: "String",
+          StringValue: JSON.stringify(options.manifest)
+        },
```

Comment:
Instead of sending the (same) manifest in every message, you could copy and import it in regeneration lambda as done in default lambda. Might even run into message limit with large manifests?

Also, you could avoid the multiple stringify/parse boilerplate by using MessageBody instead of MessageAttributes. But maybe you have some routing/filtering with attributes in mind?